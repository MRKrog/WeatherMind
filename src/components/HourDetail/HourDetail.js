import React from 'react';
import PropTypes from 'prop-types';
import { findIcon } from '../../utility/findIcon';

export const HourDetail = ({ time, icon, temperature, index }) => {

  let iconImage = findIcon(icon);

  return (
    <div className="HourDetail">
      <h4>{index === 0 ? 'Now': time}</h4>
      <div className="weatherIcon">
        <img alt={icon} src={iconImage} />
      </div>
      <h3>{temperature}</h3>
    </div>
  )
}

HourDetail.propTypes = {
  time: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temperature: PropTypes.number.isRequired,
}
