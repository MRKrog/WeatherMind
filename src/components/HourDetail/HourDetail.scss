.HourlyReport {
  display: flex;
  width: 100%;
  padding: 1em;
  overflow-x: scroll;
  background: #1d162b;
  color: #fff;
  border-top: #cccccc solid 1px;
  border-bottom: #cccccc solid 1px;
}

.HourDetail {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-right: .5em;

  h4 {
    margin: 0;
    font-weight: 100;
    font-size: .9em;
  }

  h3 {
    margin: 0;
    font-weight: 500;
    font-size: 1em;
  }

  .weatherIcon {
    width: 50px;
    height: 50px;

    .clear-night {
      background-image: url('./media/icons/clear-night.png');
      width: 50px;
      height: 50px;
      display: block;
      background-repeat: no-repeat;
    }

  }

}
