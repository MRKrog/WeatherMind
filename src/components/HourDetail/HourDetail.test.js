import React from 'react';
import { shallow } from 'enzyme';
import { HourDetail } from './HourDetail';

describe('HourDetail', () => {
  let wrapper;
  let mockIcon = "partly-cloudy-night";
  let mockIndex = 0;
  let mockTemperature = 45;
  let mockTime = "04"

  beforeEach(() => {
    wrapper = shallow(
      <HourDetail icon={mockIcon}
                  index={mockIndex}
                  temperature={mockTemperature}
                  time={mockTime}
      />
    )
  })

  it('should match the snapshot index is 0 aka "NOW" ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot for all other times', () => {
    let additionalIndex = 10
    wrapper = shallow(
      <HourDetail icon={mockIcon}
                  index={additionalIndex}
                  temperature={mockTemperature}
                  time={mockTime}
      />
    )
    expect(wrapper).toMatchSnapshot();
  });

});
