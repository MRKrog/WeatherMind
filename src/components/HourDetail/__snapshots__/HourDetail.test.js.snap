// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HourDetail should match the snapshot for all other times 1`] = `
<div
  className="HourDetail"
>
  <h4>
    04
  </h4>
  <div
    className="weatherIcon"
  >
    <img
      alt="partly-cloudy-night"
      src="partly-cloudy-night.svg"
    />
  </div>
  <h3>
    45
  </h3>
</div>
`;

exports[`HourDetail should match the snapshot index is 0 aka "NOW"  1`] = `
<div
  className="HourDetail"
>
  <h4>
    Now
  </h4>
  <div
    className="weatherIcon"
  >
    <img
      alt="partly-cloudy-night"
      src="partly-cloudy-night.svg"
    />
  </div>
  <h3>
    45
  </h3>
</div>
`;
